### PR TITLE
{AKS} Extend containerinsights `--data-collection-settings` with new fields "streams" and containerlogv2

### PR DESCRIPTION
**Pull request by wanlonghenry**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/k8s-extension/azext_k8s_extension/partner_extensions/ContainerInsights.py
Comment (ganga1980) [inline src/k8s-extension/azext_k8s_extension/partner_extensions/ContainerInsights.py:502]: two missing changes

1) add default value i.e. streams = [  "Microsoft-ContainerInsights-Group-Default"]
2) update from this 
                              "streams" : [
                                "Microsoft-ContainerInsights-Group-Default"
                            ]
to 
"streams": streams

in ContainerInsights.py file.

Comment (wanlonghenry) [reply src/k8s-extension/azext_k8s_extension/partner_extensions/ContainerInsights.py:502]: image
ContainerInsights.py is for ARC change, the aks-preview is using the code block in cli

Comment (ganga1980) [reply src/k8s-extension/azext_k8s_extension/partner_extensions/ContainerInsights.py:502]: For Arc k8s-extension, you would need to make PR against this fork https://github.com/AzureArcForKubernetes/azure-cli-extensions and branch k8s-extension/public.  Arc k8s extension will merge the changes to main branch.

**Review by ganga1980 — CHANGES_REQUESTED** [review]
This PR needs to made against  fork https://github.com/AzureArcForKubernetes/azure-cli-extensions since this for Arc k8s-extension.

**Comment by wanlonghenry** [comment]
> This PR needs to made against fork https://github.com/AzureArcForKubernetes/azure-cli-extensions since this for Arc k8s-extension.

cleaned the code for arc change, this code is now for aks-preview only

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:2839]: Queued [live test pipeline](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=75964432&view=results).

Comment (wanlonghenry) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:2839]: this is because https://github.com/Azure/azure-cli/pull/26793/files has not been merged yet, will retrigger the test after that merged.

**Comment by wanlonghenry** [comment]
@FumingZhang   I re-ran the live test, now all passed, could you review and approve when available? Thank you!

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py
Comment (wanlonghenry) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:2849]: re ran the tests, now all looks good

File: src/aks-preview/HISTORY.rst
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:17]: Please choose a new version number (and update setup.py) or move this to the pending section.

**Comment by wanlonghenry** [comment]
bumped new version, please check @FumingZhang   Thank you!

**Review by FumingZhang — APPROVED** [review]
LGTM

**Pull request merged**
